packages/grid/x-data-grid-pro/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -155,3 +158,20 @@ export const mergeStateWithRowGroupingModel =
     ...state,
     rowGrouping: { ...state.rowGrouping, model: rowGroupingModel },
   });
+
+export const setStrategyAvailability = (
+  apiRef: React.MutableRefObject<GridApiPro>,
+  disableRowGrouping: boolean,
+) => {
+  let isAvailable: () => boolean;
+  if (disableRowGrouping) {
+    isAvailable = () => false;
+  } else {
+    isAvailable = () => {
```

Comment:
`isAvailable` is a callback because when initializing the pre-processor, the state is not ready yet so we want just do something like

```ts
apiRef.current.unstable_setStrategyAvailability(
  ROW_GROUPING_STRATEGY,
  gridRowGroupingSanitizedModelSelector(apiRef).length > 0, 
);
```

And we don't want to find the value directly via the props because `gridRowGroupingSanitizedModelSelector` relies on the columns lookup. Computing the same value with the props would be heavy and in theory wrong because it would lack `hydrateColumns` processings.